administrator/components/com_admin/sql/updates/mysql/5.0.0-2023-03-17.sql

```diff
@@ -0,0 +1 @@
+DELETE FROM `#__extensions` WHERE name = 'plg_task_demotasks';
```

Comment:
```suggestion
DELETE FROM `#__extensions` WHERE `type` = 'plugin' AND `element` = 'demotasks' AND `folder` = 'task' AND `client_id` = 0;
```
Please do it this way because we do not have an index on the name column, so your statement will always result in a full table scan. We have an index on the combination of columns used in my suggested statement (which unfortunately is not a unique key as it should be): https://github.com/joomla/joomla-cms/blob/5.0-dev/installation/sql/mysql/base.sql#L148 . By using all the columns of that index in the where clause and no other columns we can be 100% sure the index will be used instead of a full table scan. Furthermore, the suggested statement is more safe because of the more precise where clause.

Reply:
P.S.: We should always do it like I suggest when we update or delete records in the extensions table because in all our code we grant that the combination of these columns is unique, see e.h. the extensions helper. The name column should not be used to identify extensions.

Reply:
Thanks!!